pkg/data/data.go

```diff
@@ -6,37 +6,68 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"strings"
 )
 
-type key string
-
-const contextDataKey key = "data"
-
+// Data holds request context data.
 type Data struct {
 	RemoteIP string
 }
 
-// ServeHTTP sets data in the request context, to be extracted with GetData.
-func ServeHTTP(w http.ResponseWriter, r *http.Request) (*http.Request, error) {
-	remoteIP, _, err := net.SplitHostPort(r.RemoteAddr)
+type key int
+
+const contextDataKey key = iota
+
+// GetRemoteIP extracts the remote IP from the request, optionally using a custom header.
+func GetRemoteIP(r *http.Request, sourceIPHeader string) (string, error) {
```

Comment:
For this I had invision to have a configuration more like [Traefik's source criterion](https://doc.traefik.io/traefik/middlewares/http/ratelimit/#sourcecriterionipstrategy) which allows choosing which IP in the header is considered.
This will not ban your firewall, for example !

WDYT ?

Reply:
Yes, I agree. I listed this as an improvement but I guess I could try to implement it.